lib/ansible/modules/database/postgresql/pg_hba.py

```diff
@@ -0,0 +1,769 @@
+    return int_to_ipv4((2 ** base - 1) * 2 ** (32 - base))
+
+
+def ipv6_to_int(ip):
+    normalized = ip
+    if '.' in normalized:
+        # Normalize: Replace ipv4 part for ipv6 equivalent
+        m = ipv4part_re.search(normalized)
+        ipv6part = ''.join('%02x' % int(i) for i in m.group(0).split('.'))
+        ipv6part = ipv6part[:4] + ':' + ipv6part[4:]
+        normalized = normalized.replace(m.group(0), ipv6part)
+    if '::' in ip:
+        # Normalize: Replace :: for correct number of 0000 parts
+        missing = 9 - normalized.count(':')
+        normalized = normalized.replace('::', ":".join([''] + ['0000'] * missing + ['']))
+        normalized.strip(':')
+        normalized = normalized.replace('::', ':')
+    parts = normalized.split(':')
+    if len(parts) < 8:
+        raise IPError('IPv6 seems to consist of too less parts')
+    elif len(parts) > 8:
+        raise IPError('IPv6 seems to consist of too much parts')
+    # Normalize: Every part should have 4 digits
+    for i in range(parts):
+        if len(parts[i]) != 4:
+            part = '0000' + parts[i]
+            parts[i] = part[-4:]
+    return int(parts.replace(':', ''), 16)
+
+
+def int_to_ipv6(i):
+    try:
+        rest = int(i)
+    except:
+        raise IPError("{0} is not an integer.".format(i.__repr__()))
+
+    # Split into hex parts
+    ipv6 = []
+    for i in range(8):
+        part = rest % 16**4
+        rest = rest / 16**4
+        ipv6.append(str(part))
+
+    # join with ':' as seperator
+    ipv6 = ':'.join(ipv6[::-1])
+
+    # Find largest repetion of zero fields and replace by '::'
+    obsoletes = [m.group(0) for m in ipv6_obs_re.finditer(ipv6)]
+    if len(obsoletes) > 0:
+        obsoletes = sorted(obsoletes, key=len)
+        largest_obsolete = obsoletes[-1]
+        ipv6 = ipv6.replace(obsoletes, '::', 1)
+
+    # Strip leading zeros per field
+    parts = ipv6.split(':')
+    for i in range(len(parts)):
+        part = parts[i]
+        if len(part) == 0:
+            continue
+        part = part.lstrip('0')
+        if len(part) == 0:
+            part = '0'
+        parts[i] = part
+    return ':'.join(parts)
+
+
+def prefix_to_ipv6netmask(base):
+    if isinstance(base, str):
+        base = base.replace('/', '')
+    try:
+        base = int(base)
+    except:
+        raise IPError("invalid numeric expression for ipv6 network base {}".format(base))
+    return int_to_ipv6((2 ** base - 1) * 2 ** (128 - base))
+
+
+PgHbaMethods = ["trust", "reject", "md5", "password", "gss", "sspi", "krb5", "ident", "peer", "ldap", "radius", "cert", "pam"]
+PgHbaTypes = ["local", "host", "hostssl", "hostnossl"]
+PgHbaOrders = ["sdu", "sud", "dsu", "dus", "usd", "uds"]
+PgHbaHDR = ['type', 'db', 'usr', 'src', 'mask', 'method', 'options']
+
+split_re = re.compile('\s+')
+
+
+# See http://stackoverflow.com/questions/53497/regular-expression-that-matches-valid-ipv6-addresses for more info...
+IPV4SEG = '(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])'
```

Comment:
[ipaddress](https://docs.python.org/3/library/ipaddress.html) Python module should be used instead in order to check/handle IP addresses. This module is included in Python 3.3 and backport exists for previous versions.

Reply:
Modified in latest commit.